include/deal.II/base/point.h

```diff
@@ -646,6 +463,57 @@ inline void Point<dim, Number>::serialize(Archive &ar, const unsigned int)
 
 /*--------------------------- Global functions: Point -----------------------*/
 
+/**
+ * Add an offset given as Tensor<1,dim,Number> to a point.
+ *
+ * @note This function can also be used in @ref GlossDevice "device" code.
+ */
+template <int dim, typename Number>
+DEAL_II_CXX20_REQUIRES(dim >= 0)
+constexpr DEAL_II_HOST_DEVICE Point<dim, Number>
+operator+(const Point<dim, Number> &p, const Tensor<1, dim, Number> &offset)
+{
+  Point<dim, Number> tmp{p};
+  tmp += offset;
+  return tmp;
+}
```

Comment:
Ah, I see -- you are replacing the member function by a global function. It's not clear to me why this is advantageous, though. What is the rationale?

Reply:
Ah, I remember some posts, say [this stackoverflow post](https://stackoverflow.com/questions/4622330/operator-overloading-member-function-vs-non-member-function), suggest that it is better to write the binary operator as a non-member function considering the invoke order.

Here it does not matter too much, since we have Tensor operator in the bottom anyway. But it is a better convention from my perspective.

Reply:
I think this is perhaps debatable. The stackoverflow post you point out is of course correct if you want an operator where the *left* operand is a non-class type, but that isn't the case here.

There is, however, a good reason to leave the code as it is: The documentation of this function will clearly show up as part of the documentation of the class. With global functions, doxygen will create a page for the function, but it may not be linked to the documentation of the `Point` class.

Reply:
You are right. There's no difference in this situation but since `Tensor` also leave its `operator+-*/` as a global function, maybe we should also do it with `Point`?

We could use `@relates` to hint doxygen to link `operator+-` and it will only appear on that page just like other functions. What do you think?

Reply:
Yes, if you can make it look essentially like `Tensor`, then I'm in favor :-)